bundles/org.openhab.binding.telegram/src/main/java/org/openhab/binding/telegram/internal/TelegramHandler.java

```diff
@@ -143,90 +148,115 @@ public void initialize() {
                 }
-                return UpdatesListener.CONFIRMED_UPDATES_ALL;
+                updateChannel(LASTMESSAGETEXT,
+                        lastMessageText != null ? new StringType(lastMessageText) : UnDefType.NULL);
+                updateChannel(LASTMESSAGEDATE,
+                        lastMessageDate != null
+                                ? new DateTimeType(ZonedDateTime
+                                        .ofInstant(Instant.ofEpochSecond(lastMessageDate.intValue()), ZoneOffset.UTC))
+                                : UnDefType.NULL);
+                updateChannel(LASTMESSAGENAME, (lastMessageFirstName != null || lastMessageLastName != null)
+                        ? new StringType((lastMessageFirstName != null ? lastMessageFirstName + " " : "")
+                                + (lastMessageLastName != null ? lastMessageLastName : ""))
+                        : UnDefType.NULL);
+                updateChannel(LASTMESSAGEUSERNAME,
+                        lastMessageUsername != null ? new StringType(lastMessageUsername) : UnDefType.NULL);
+                updateChannel(CHATID, chatId != null ? new StringType(chatId.toString()) : UnDefType.NULL);
+                updateChannel(REPLYID, replyId != null ? new StringType(replyId) : UnDefType.NULL);
+            }
+            return UpdatesListener.CONFIRMED_UPDATES_ALL;
         }, exception -> {
-                if (exception != null) {
-                    logger.warn("Telegram exception: {}", exception.getMessage());
-                    if (exception.response() != null) {
-                        BaseResponse localResponse = exception.response();
-                        if (localResponse.errorCode() == 401) {
-                            logger.error("Bot token invalid, disable thing {}", getThing().getUID());
-                            localBot.removeGetUpdatesListener();
-                            updateStatus(ThingStatus.OFFLINE);
-                        }
+            if (exception != null) {
+                if (exception.response() != null) {
+                    BaseResponse localResponse = exception.response();
+                    if (localResponse.errorCode() == 401) { // unauthorized
+                        cancelThingOnlineStatusJob();
+                        localBot.removeGetUpdatesListener();
+                        updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                                "Unauthorized attempt to connect to the Telegram server, please check if the bot token is valid");
+                        return;
                     }
+                }
+                if (exception.getCause() != null) { // cause is only non-null in case of an IOException
+                    cancelThingOnlineStatusJob();
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, exception.getMessage());
+                    delayThingOnlineStatus();

```

Comment:
Are errors repeated every <10sec? Otherwise the thing would go back to online with this function call, wouldn't it?

Reply:
Yes, let's say the connection is broken an an IOException is reported, the Telegram library puts the thread to [sleep](https://github.com/pengrad/java-telegram-bot-api/blob/82049f1ec822d87ef3ff5c8a29116cd065b9a13e/library/src/main/java/com/pengrad/telegrambot/impl/UpdatesHandler.java#L88) for 100ms (default value) before it creates a new request (that might fail again with an IOException) and thus produces a lot of messages in the log.

If you like the previous "optimistic" solution better (Thing went ONLINE immediately, and only with an error went to OFFLINE), then I'm also happy to keep the "Status quo". I think the most important change is that the updateStatus() call doesn't produce more log entries when the Thing was already set to OFFLINE.

Reply:
The 100ms feels a bit strange, but when this is the default for the library ...

Reply:
I can confirm this and I also find it strange. Mainly this happens when internet is down or Telegram servers are down, I don't know why 100ms is default there...

Reply:
It's possible to increase that number. But would it make a big difference to increase it? (ok, it would produce a bit less load in case of a connection error). But I don't know what a "suitable" value would be here.